feat(docs): Update monitoring docs to use observability plugin

jasonmcintosh opened the pull request.

jasonmcintosh committed:
Added extra docs around observability plugin

jasonmcintosh committed:
Add info about grafana dashboards

alchen99 commented on content/en/docs/spinnaker-install-admin-guides/prometheus-monitoring.md:
```diff
@@ -53,7 +61,190 @@ Access the Prometheus web interface by using the `kubectl port-forward` command.
+      - spin-deck
+      - spin-deck-custom
+      - spin-redis
+      - spin-terraformer
```
@jasonmcintosh Need to add `spin-dinghy` to this list.

jasonmcintosh replied:
Should be done ;)

jasonmcintosh committed:
Added one more exclude for dinghy

317brian committed:
copyedits and update breaking change in release notes

317brian commented on content/en/docs/spinnaker-install-admin-guides/prometheus-monitoring.md:
````diff
@@ -53,141 +62,190 @@ Access the Prometheus web interface by using the `kubectl port-forward` command.
 
-```bash
-halyard-0:~ $ hal config metric-stores prometheus enable
+<!-- Spinnaker issue discussing management endpoints: https://github.com/spinnaker/spinnaker/issues/3883--> 
````
I commented out this line cause I wasn't sure what we wanted the end user to get out of it. Can add back in based on your input @jasonmcintosh

317brian commented on content/en/includes/bc-metrics-name.md:
```diff
@@ -0,0 +1,18 @@
+### Spinnaker metrics
```
Replaced the copy/paste in the release notes w/ a reusable snippet.

317brian commented on content/en/docs/spinnaker-install-admin-guides/prometheus-monitoring.md:
```diff
@@ -5,18 +5,27 @@ description: >
   Monitoring Spinnaker using Prometheus and Grafana
 ---
 
-{{% alert title="Warning" color="warning" %}}There is a known issue with metric names in version 2.20.x. Until this issue is resolved, any dashboards created from the instructions on this page will not work. For more information, see the release notes for your version, such as [2.20.5]({{< ref "armoryspinnaker_v2-20-5#spinnaker-metrics" >}}). {{% /alert %}}
```
Removed the warning since the new docs cover what's needed for 2.20+

317brian reviewed (approved):
Thanks for putting this together @jasonmcintosh. I made some copyedits and updated the release note. Had one question about line 80 and the link to the spinnaker issue. If it's not a big deal 👍

christosarvanitis committed:
Resolving conflicts

christosarvanitis committed:
Update Observability Plugin example to 1.1.3

christosarvanitis committed:
Updating with Observability plugin details

mrusan commented on .gitignore:
```diff
@@ -1,4 +1,4 @@
-
+.idea
```
We should probably exclude files/directories not generated by the code in this repository from `.gitignore`. Maintaining own `.gitignore_global` for OS/IDE-specific local files is best practice.

317brian committed:
remove monitoring daemon info at the bottom. fix broken images. make copyedits

317brian committed:
add back monitoring daemon to ask about it

317brian commented on content/en/docs/armory-admin/prometheus-monitoring.md:
```diff
@@ -54,23 +59,241 @@ Access the Prometheus web interface by using the `kubectl port-forward` command.
+
+![Grafana Dashboard](/images/grafana-2.png)
+
+## Configure monitoring using the monitoring-daemon
```
@christosarvanitis can we remove everything from here until the end of the file since monitoring-daemon is no longer supported by Armory and we recommend the Observability plugin?

christosarvanitis replied:
@brian-armory sure, if you prefer to completely remove any mention to the monitoring-daemon then we can remove

317brian committed:
remove monitoring daemon info since it is no longer supported
This reverts commit 0222f97e198da33374cadc4aa33f6e54a839c2df.

karlomendozaarmory commented on content/en/docs/armory-admin/prometheus-monitoring.md:
```diff
@@ -54,92 +59,170 @@ Access the Prometheus web interface by using the `kubectl port-forward` command.
+kind: ServiceAccount
+metadata:
+  namespace: monitoring
+  name: prometheus
```
Prometheus release-0.4 ServiceAccount is called prometheus-k8s. This makes prometheus fail for permissions. Might want to add a note about this.

christosarvanitis replied:
Actually we dont even require the whole Grant Prometheus RBAC permissions section. By installing the release-0.4 of prometheus-kube the RBAC are there by default. @brian-armory  Should we remove it?

karlomendozaarmory commented on content/en/docs/armory-admin/prometheus-monitoring.md:
```diff
@@ -54,92 +59,170 @@ Access the Prometheus web interface by using the `kubectl port-forward` command.
+  name: prometheus
+subjects:
+  - kind: ServiceAccount
+    name: prometheus
```
Prometheus release-0.4 ServiceAccount is called prometheus-k8s. This makes prometheus fail for permissions. Might want to add a note about this.

christosarvanitis replied:
Actually we dont even require the whole Grant Prometheus RBAC permissions section. By installing the release-0.4 of prometheus-kube the RBAC are there by default. @brian-armory Should we remove it?

karlomendozaarmory commented on content/en/docs/armory-admin/prometheus-monitoring.md:
```diff
@@ -54,92 +59,170 @@ Access the Prometheus web interface by using the `kubectl port-forward` command.
+      - spin-dinghy
+    matchLabels:
+      app: spin
+    namespaceSelector:
```
namespaceSelector does not exists in my prometheus versions, I just removed this part and it worked.

christosarvanitis replied:
👍 removing it

karlomendozaarmory commented:
Dinghy and Terraformer don't send metrics are we aware/happy about this?
We should at least add a note about that in the doc.

karlomendozaarmory commented on content/en/docs/armory-admin/prometheus-monitoring.md:
```diff
@@ -177,18 +262,16 @@ Access the Grafana web interface via http://localhost:3000 and use the default g
 To import the sample dashboards, perform the following steps:
 
-1. Git clone this repo to your local workstation: (https://github.com/spinnaker/spinnaker-monitoring)
+1. Git clone this repo to your local workstation: (https://gitlab.com/uneeq-oss/spinnaker-mixin)
```
Are going to suggest using this dashboards? We now have the plugin version of the old dashboards at:
https://github.com/armory-io/spin-monitoring-dashboards/pull/2

christosarvanitis replied:
This is inline with the previous version of the document. We were referring to the monitoring-daemon OSS dashboards with a statement that Armory provides additional dashboards to our customers. @brian-armory thoughts?

317brian replied:
@karlomendozaarmory 's link is to the internal Armory GitHub, so that wouldn't be helpful to customers. For now, let's link to the public examples that @christosarvanitis added. 

I know there are extra ones we provide to customers only, and IIRC those aren't linked to directly in this doc.

christosarvanitis committed:
Fixing conflict with master

christosarvanitis committed:
Removing namespaceSelector

christosarvanitis committed:
Removing namespaceSelector

317brian committed:
add note about dinghy and terraformer

317brian commented:
> Dinghy and Terraformer don't send metrics are we aware/happy about this?
> We should at least add a note about that in the doc.

Added a line that says this doc doesn't cover Dinghy and Terraformer

Commit:
Added several comments

* About name: prometheus possibly being name: prometheus-k8s
  depending on version of prometheus-operator installed
* To ServiceMonitor spinnaker-all-metrics explaining release
* To ServiceMonitor spinnaker-internal-metrics explaining
  use of scheme vs targetPort

alchen99 reviewed (commented):
@brian-armory I've added a commit with some changes. If it looks good to you then I'm good to go.

317brian committed:
copyedits

317brian reviewed (approved):
Thanks for working on this everybody!

317brian committed:
copyedits
